common_util/src/partitioned_lock.rs

```diff
@@ -100,6 +101,50 @@ impl<T> PartitionedMutex<T> {
     }
 }
 
+#[derive(Debug)]
+pub struct PartitionedMutexAsync<T> {
+    partitions: Vec<tokio::sync::Mutex<T>>,
+    partition_mask: usize,
+}
+
+impl<T> PartitionedMutexAsync<T> {
+    pub fn new<F>(init_fn: F, partition_bit: usize) -> Self
```

Comment:
Maybe this init_fn could be improved to something like
```
Fn(usize) -> T
```
The param is total_capacity, then upper layer don't need to calculate themselves. 

But in order to make this PR clean, you can refactor this in next PR.